linalg/sparsemat.hpp

```diff
@@ -151,8 +170,15 @@ class SparseMatrix : public AbstractSparseMatrix
    /// Create a SparseMatrix with diagonal @a v, i.e. A = Diag(v)
    SparseMatrix(const Vector & v);
 
-   // Runtime option to use cuSPARSE. Only valid when using a CUDA backend.
-   void UseCuSparse(bool useCuSparse_ = true) { useCuSparse = useCuSparse_;}
+   /** @brief Runtime option to use cuSPARSE or hipSPARSE. Only valid when using
+       a CUDA or HIP backend.
+
+       @note This option is enabled by default, so typically one would use this
+       method to disable the use of cuSPARSE/hipSPARSE. */
+   void UseGPUSparse(bool useGPUSparse_ = true) { useGPUSparse = useGPUSparse_;}
+   /// Deprecated equivalent of UseGPUSparse().
+   MFEM_DEPRECATED
```

Comment:
Would it be possible to have
```c++
MFEM_DEPRECATED("Deprecated equivalent of UseGPUSparse().");
```
in order to print the message at compilation?

Reply:
Adding messages to `MFEM_DEPRECATED` is a big change to make in this PR (it will affect a lot of places), so I think it should be done in a separate PR.